frontend/packages/core/src/Input/select.tsx

```diff
@@ -201,28 +201,33 @@ const Select = ({
   options,
   onChange,
 }: SelectProps) => {
-  if (options?.length === undefined || options?.length === 0) {
-    return null;
-  }
-
-  const defaultIdx = defaultOption < options.length ? defaultOption : 0;
+  const [defaultIdx] = React.useState(
```

Comment:
why is this a hook?

Reply:
To ensure that the following dependent states are properly updated in order

Reply:
i don't follow, would like to understand for my own personal benefit. what happens then if this is not a hook vs if it is?

Reply:
Ah nevermind, ignore this :)